### ISSUE TITLE
readOGR warning message 

### ISSUE THREAD

**Issue by lucymerobinson** [lucymerobinson/CCAMLRGIS#20]
a recent change to rgdal in v1.2-18 now results in the following warning message when loading GeoJSON data
e.g. 
SSMUs <- load_SSMUs("GEOJSON")
Warning messages:
1: In normalizePath(dsn) :
  path[1]="https://gis.ccamlr.org/geoserver/gis/ows?service=WFS&version=1.0.0&request=GetFeature&typeName=gis:ssmus&outputFormat=json": No such file or directory

This issue seems similar to the issue reported [here ](https://github.com/ropensci/geojsonio/issues/130)
I could suppress the warning or I could substitute rgdal::readOGR with sf:: st_read (as suggested in the issue post referred to above). What do you think @mdsumner ?

**Comment by lucymerobinson** [comment]
example code inside load_SSMUs function using sf function 
SSMU_data<- as(st_read(ccamlrgisurl, quiet = TRUE), "Spatial")

**Comment by mdsumner** [comment]
I don't know sorry, I'm not able to read these sources at all today - but that might be my system. 

Is it just a warning, or is the read also failing? If a warning can wrap the read with

```R
op <- options(warn = -1)
on.exit(options(op))  ## reset the warning setting when this function returns
x <- readOGR(etc)

```

**Comment by mdsumner** [comment]
The geojsonio approach doesn't look very promising, there's no way rgdal should assume a source is a filename, so I'm surprised if that's the case. I'll try to follow up. 

Possibly could simple obtain the GeoJSON text by reading it raw, and actually creating a file and reading - or using a more generic GeoJSON reader (like geojsonio, but there are others)

**Comment by lucymerobinson** [comment]
It's reading and is just a warning message, but I also had to remove readLines( url ,warn=F) to get readOGR to work. I think I will just go with suppressWarnings option for now.

**Comment by mdsumner** [comment]
Cool, that's really a sucky bug for rgdal to assume a normalizeable path for a dsn!

**Comment by lucymerobinson** [comment]
I am hoping it will be fixed in the next version!

**Comment by lucymerobinson** [comment]
Hi @mdsumner I thought I had fixed the readOGR issue, but while it works on my mac, my fix doesn't appear to work on windows. If you have a chance to look into this that would be much appreciated.

**Comment by lucymerobinson** [comment]
I replaced readOGR with read_sf and it worked on both windows and mac platforms that I have access to, but builds in appveyor and travis are failing so more work to be done by the looks of it

**Comment by lucymerobinson** [comment]
on travis the issues seems to be with an sf dependency package ‘udunits2’

**Comment by mdsumner** [comment]
I don't think pulling sf in is a good idea right now, downloading the file locally should work - though the layername may well be different in different builds of rgdal. (that might be the problem ultimately)

Maybe something like
```R
u <- "https://gis.ccamlr.org/geoserver/gis/ows?service=WFS&version=1.0.0&request=GetFeature&typeName=gis:ssmus&outputFormat=json"
temp <- sprintf("%s.json", tempfile())
download.file(u, temp, mode = "wb")
readOGR(temp, ...)

```

I can't test it right now though, have some other problem :|

**Comment by lucymerobinson** [comment]
this issue with readOGR seems to be fixed in rgdal v1.3.2 @mdsumner. The build in Appveyor is not passing (but travis is), but I can install on a local windows machine and I suspect its not related to this issue so I'm closing this.
